### ISSUE TITLE
Included custom data field sets are now nullable

### ISSUE THREAD

**Issue by vestrel00** [vestrel00/contacts-android#324]
## Problem

Due to changes in #321, the following custom data components now take in nullable custom data field sets (e.g. `Set<AbstractCustomDataField>?`).

When upgrading to the latest release, library users that define their own custom data will run into compiler errors in implementations of;

- `AbstractCustomDataCursor`
- `AbstractCustomDataEntityMapper.Factory`
- `AbstractCustomDataOperation`
- `AbstractCustomDataOperation.Factory`

## Solution

Notify library users of this breaking change!

- [ ] Close this issue along with #321
- [x] Provide migration guide.

**Comment by vestrel00** [comment]
## Migration guide (to be included in release notes)

Fix compile errors by making `Set<AbstractCustomDataField>` nullable -> `Set<AbstractCustomDataField>?`

For example, change this...

```kotlin
internal class HandleNameMapperFactory {
    override fun create(
        ... includeFields: Set<HandleNameField>
    ) = HandleNameMapper(HandleNameDataCursor(cursor, includeFields))
}
```

to this...

```kotlin
internal class HandleNameMapperFactory {
    override fun create(
        ... includeFields: Set<HandleNameField>?
    ) = HandleNameMapper(HandleNameDataCursor(cursor, includeFields))
}
```

**Comment by vestrel00** [comment]
This is included in release [0.3.2](https://github.com/vestrel00/contacts-android/releases/tag/0.3.2).
